### DISCUSSION TITLE
v1.0 规划

### DISCUSSION THREAD

**Discussion by naiba** [nezhahq/nezhahq.github.io#35] [Ideas]
- Agent
   - Agent 端本地生成一个 ID (?UUID) 加上一个统一的服务器端密钥与服务器端进行通信。解决必须手动一个一个添加的痛点。
   - Agent 配置全部写到配置文件，方便统一更新
   - [?待定] Agent 自启动 从脚本安装改为内置 ( https://github.com/kardianos/service )
   - [?] GPU https://github.com/jaypipes/ghw#gpu
   - [?] 温度 暂无可用轮子
- Dashboard
   - 前后端分离，新的 UI https://www.figma.com/file/wmhiIuo2iqWp78yV3OPeqI/Nezha-Monitor?type=design
   - 合并 grpc/http 端口
   - 配置项全部存入数据库，避免 重装/更新 覆盖掉配置
   - 批量删除节点
   - [?待设计] 历史记录
   - [?待设计] 服务报警/服务器指标报警 变更为更灵活的执行自定义脚本，或利用 ( https://pkg.go.dev/plugin ) 实现动态加载功能，服务器端可配置启用哪些功能
   - [?待设计] 抽象数据库允许功能模块 插入/读取 数据

**Comment by spiritLHLS** [comment]
提一个节点批量删除的功能需求，之前有批量改分区可惜没有批量删的

**Reply by naiba** [reply]
采纳

**Comment by zaxigia** [comment]
希望可以添加个`过期时间`字段

**Comment by evanfury** [comment]
能不能在监控页面（服务器状态展示页面）加一个地图？在地图上面pin服务器 然后不同颜色代表服务器状态？🤩

**Reply by xsmzdd** [reply]
<img width="1436" alt="165958225-25fc8fda-5798-42f8-bac5-72d778c0bab5" src="https://github.com/nezhahq/nezhahq.github.io/assets/111579783/201e5f94-4449-47e3-8236-394bebf3bf65">

类是这样,再按你说的加入ping,按延迟.比如0-100显示绿色,100-200显示黄色,200-300显示橙色,300以上显示红色

**Comment by lemongad** [comment]
添加一个搜索功能与自动分组功能，现在后台看起来很乱，不同组的杂乱在一起。搜索功能能帮助快速找到要管理的机器。

**Comment by lemongad** [comment]
添加webftp的功能，方便与管理的机器随时上传下载文件

**Comment by xsmzdd** [comment]
后台排序可以增加几个,比如名称和服务器分组
服务页面中排序问题,也是只能按编号排序,而且对应服务器编号还不能修改.强迫症!
(主要是服务器多,后期添加,修改一个就要全部删除再一个个导入)
![QQ截图20230626204021](https://github.com/nezhahq/nezhahq.github.io/assets/111579783/0519044e-8507-42b2-94c2-0ef4ec7f5293)

**Reply by spiritLHLS** [reply]
老实说，不只是排序问题，如果服务器太多，需要找到分组还得搜索一番，如果分组不同可以选择显示哪几个分组在后台显示就更好了(或者分页显示，页面太长了)

**Comment by Hothua** [comment]
希望后台服务列表增加自定义列的功能和服务器到期时间/费用的功能

比如
**对游客隐藏 | 启用DDNS | 启用DDNS IPv4 | 启用DDNS IPv6 | DDNS域名**
按需启用，不然折行看着太难受
